packages/influx-metrics-tracker/src/helpers/exponential-backoff.ts

```diff
@@ -2,23 +2,34 @@ import { Logger } from '@ovotech/winston-logger';
 
 const ONE_SECOND = 1000;
 
+function sleep(sleepTimeMs: number) {
+  return new Promise(resolve => {
+    setTimeout(() => {
+      resolve({});
+    }, sleepTimeMs);
+  });
+}
+
 export async function executeCallbackOrExponentiallyBackOff(
   callback: (...args: any[]) => void,
   logger: Logger,
-  timer = ONE_SECOND,
+  timer = 0,
+  di_runAllTimers = () => {},
 ): Promise<void> {
   try {
     logger.info('Executing Influx Metrics Tracker batch callback');
+    const sleepTimer = sleep(timer);
+    di_runAllTimers();
+    await sleepTimer;
```

Comment:
should this be run after the sleep timer? otherwise theres no delay for first and second time it's called

Reply:
If its run after the await then then `jest.runAllTimers()` never gets called until after the function exits at which point its too late. In this case the timers would run at their normal time instead of benefitting from the `runAllTimers` feature. Does this make sense?